langworld_db_data/featureprofiletools/value_renamer.py

```diff
@@ -0,0 +1,113 @@
+        new_value_name: str,
+    ) -> None:
+        """
+        Renames value with given ID in all feature profiles and inventory of listed values.
+        Also works with elementary values in multiselect features.
+        """
+        files = list(self.input_feature_profiles_dir.glob("*.csv"))
+        for file in files:
+            print(f"Processing {file.name}")
+            self._update_one_feature_profile(
+                id_of_value_to_rename=id_of_value_to_rename,
+                new_value_name=new_value_name,
+                input_file=file,
+                output_dir=self.output_feature_profiles_dir,
+            )
+
+        if not self.output_inventories_dir.exists():
+            self.output_inventories_dir.mkdir(parents=True, exist_ok=True)
+        self._update_features_listed_values(
+            id_of_value_to_rename=id_of_value_to_rename,
+            new_value_name=new_value_name,
+            input_file=self.input_inventories_dir / "features_listed_values.csv",
+            output_file=self.output_inventories_dir / "features_listed_values.csv",
+        )
+
+    @staticmethod
+    def _update_one_feature_profile(
+        id_of_value_to_rename: str,
+        new_value_name: str,
+        input_file: Path,
+        output_dir: Path,
+    ) -> None:
+
+        number_of_replacements = 0
+        data_from_file = read_dicts_from_csv(input_file)
+        data_to_write = []
+        for line in data_from_file:
+            if id_of_value_to_rename not in line["value_id"]:
+                data_to_write.append(line)
+                continue
+            line_to_write = line.copy()
+            # After this clause, only lines with the target value are considered (they may contain other values too)
+            if "&" in line["value_id"]:
+                print("Found match in combined value in " + input_file.name)
+                combined_value_ids = line["value_id"].split("&")
+                target_value_index = combined_value_ids.index(id_of_value_to_rename)
+                combined_value_names = line["value_ru"].split("&")
+                combined_value_names.pop(target_value_index)
+                combined_value_names.insert(target_value_index, new_value_name)
+                line_to_write["value_ru"] = "&".join(combined_value_names)
+                number_of_replacements += 1
+                data_to_write.append(line_to_write)
+                print("Changed " + line["value_ru"] + " to " + line_to_write["value_ru"])
+                continue
+            # After this clause, only those lines are considered which contain the target value only
+            print(f"Found exact match in {input_file.name}")
+            line_to_write["value_ru"] = new_value_name
+            number_of_replacements += 1
+            data_to_write.append(line_to_write)
+            print("Changed " + line["value_ru"] + " to " + new_value_name)
+        print("Replacements made in this file:" + str(number_of_replacements))
```

Comment:
by the way, an f-string here will save you a call to `str()`

Reply:
Rewritten this line with an f-string